modules/infra/submodules/cost-usage-report/glue.tf

```
@@ -0,0 +1,123 @@
+# An AWS Glue database
+# An AWS Glue crawler
+# Two Lambda functions
+
+resource "aws_glue_catalog_database" "aws_cur_database" {
+  description = "Contains CUR data based on contents from the S3 bucket '${aws_s3_bucket.cur_report.bucket}'"
+  name        = local.aws_glue_database
+  catalog_id  = local.aws_account_id
+}
+
+
+resource "aws_glue_security_configuration" "lambda_config" {
+  name = "${var.deploy_id}_lambda_security_config"
+
+  encryption_configuration {
+    cloudwatch_encryption {
+      cloudwatch_encryption_mode = "SSE-KMS"
+      kms_key_arn                = local.kms_key_arn
+    }
+
+    job_bookmarks_encryption {
+      job_bookmarks_encryption_mode = "CSE-KMS"
+      kms_key_arn                   = local.kms_key_arn
+    }
+
+    s3_encryption {
+      s3_encryption_mode = "SSE-KMS"
+      kms_key_arn        = local.kms_key_arn
+    }
+  }
+}
+
+
+resource "aws_glue_crawler" "aws_cur_crawler" {
+  name          = local.cur_crawler
+  description   = "A recurring crawler that keeps your CUR table in Athena up-to-date."
+  database_name = aws_glue_catalog_database.aws_cur_database.name
+  role          = aws_iam_role.aws_cur_crawler_component_function_role.name
+
+  s3_target {
+    path = "s3://${aws_s3_bucket.cur_report.bucket}/${var.cost_usage_report.s3_bucket_prefix}/${var.cost_usage_report.report_name}/${var.cost_usage_report.report_name}"
+    exclusions = [
+      "**.json",
+      "**.yml",
+      "**.sql",
+      "**.csv",
+      "**.gz",
+      "**.zip",
+    ]
+  }
+
+  schema_change_policy {
+    delete_behavior = "DELETE_FROM_DATABASE"
+    update_behavior = "UPDATE_IN_DATABASE"
+  }
+
+  security_configuration = aws_glue_security_configuration.lambda_config.name
+
+  tags = var.tags
+
+  depends_on = [
+    aws_glue_catalog_database.aws_cur_database,
+    aws_iam_role.aws_cur_crawler_component_function_role,
+    aws_s3_bucket.cur_report
+  ]
+}
+
+resource "aws_glue_catalog_table" "aws_cur_report_status_table" {
+  name          = local.report_status_table_name
+  database_name = aws_glue_catalog_database.aws_cur_database.name
+  table_type    = "EXTERNAL_TABLE"
+  catalog_id    = local.aws_account_id
+
+  parameters = {
+    EXTERNAL              = "TRUE"
+    "parquet.compression" = "SNAPPY"
+  }
+
+  storage_descriptor {
+    location      = "s3://${aws_s3_bucket.cur_report.bucket}/${var.cost_usage_report.s3_bucket_prefix}/${var.cost_usage_report.report_name}/${local.report_status_table_name}/"
```

nitpick, I have found that the `/` makes room for errors, perhaps, 
```suggestion
    location      = "s3://${${join("/",[aws_s3_bucket.cur_report.bucket,var.cost_usage_report.s3_bucket_prefix,var.cost_usage_report.report_name,local.report_status_table_name])}}/"
```